### PR TITLE
Release environment must use jenkins-infra repository

### PR DESCRIPTION
**Pull request by olblak**
* Switch release.ci.jenkins.io to lts version and automate its upgrade when a new docker image is available based on digest
* Rename jenkins.yaml configuration file to jenkins-release.yaml to be more explicit on the content
* Fix release.ci.jenkins.io packaging job to use jenkins-infra/release instead of olblak/release

### CODE REVIEW

File: updateCli.d/jenkins-release.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+source:
+  kind: dockerDigest
+  spec:
+    image: "jenkins/jenkins"
+    tag: "lts"
```
Comment (timja) [inline updateCli.d/jenkins-release.yaml:5]: should this be?:
```suggestion
    tag: "lts-jdk11"
```

Comment (olblak) [reply updateCli.d/jenkins-release.yaml:5]: Yes, nice catch then we also have to update the digest

File: config/default/jenkins-release.yaml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 clusterAdminEnabled: false
 jenkins:
     master:
+        image: jenkins/jenkins@sha256
+        imageTag: af0110d8f70d2fbaf9429c321dc4e3bc5aae12f708c655f4018e7ed4be0974f5
```
Comment (olblak) [inline config/default/jenkins-release.yaml:5]:
```suggestion
        imageTag: 973a0d1dd45f0706ac720c50f14f8ea7de6a80583cc0100af76d4288a7d89714
```

Comment (olblak) [reply config/default/jenkins-release.yaml:5]: this digest correspond to the latest jenkins/jenkins:lts-jdk11

**Pull request merged**